src/qibolab/instruments/rfsoc.py

```diff
@@ -262,12 +365,31 @@ def classify_shots(self, i_values: List[float], q_values: List[float], qubit: Qu
             return [shots]
         return shots
 
+    def base_recursion_layer_sweep(
+        self,
+        qubits: List[Qubit],
+        sequence: PulseSequence,
+        or_sequence: PulseSequence,
+        execution_parameters: ExecutionParameters,
+    ) -> Dict[str, Union[IntegratedResults, SampleResults]]:
+        """Last recursion layer, if no sweeps are present"""
+        res = self.play(qubits, sequence, execution_parameters)
+        newres = {}
+        serials = [pulse.serial for pulse in or_sequence.ro_pulses]
+        for idx, key in enumerate(res):
+            if idx % 2 == 1:
+                newres[serials[idx // 2]] = res[key]
```

Comment:
This is sufficiently complex since, in case in is an odd index, you're saving the result in the serial listed at (n - 1)/2 position.
And, from the point of view of this function, indices and serials are not connected.

Some local documentation would be great. Even better if in the docstring.